core/src/main/java/org/apache/iceberg/PartitionData.java

```diff
@@ -191,8 +191,10 @@ public boolean equals(Object o) {
   @Override
   public int hashCode() {
     Hasher hasher = Hashing.goodFastHash(32).newHasher();
-    Stream.of(data).map(Objects::hashCode).forEach(hasher::putInt);
-    partitionType.fields().stream().map(Objects::hashCode).forEach(hasher::putInt);
+    if (size > 0) {
```

Comment:
What about returning a static value early rather than creating a hasher and not adding data to it?

Reply:
Sound good, but I don't know which static value I should choose to prevent collision with others. I use `Hashing.goodFastHash(32).newHasher().hash().hashCode()` to calculate and the result is 517856167, I don't know if it is suitable. Can you give me some advice?